test-framework/common/src/main/java/io/quarkus/test/common/GroovyClassValue.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.test.common;
+
+/**
+ * {@link ClassValue} are used in Groovy which causes memory leaks if not properly cleaned, but unfortunately, they are
+ * very complex to clean up, especially the {@link ClassValue} corresponding to system classes that must be cleaned too
+ * to avoid memory leaks moreover if not cleaned wisely errors of type {@code MissingMethodException} can be thrown, so
+ * we had better to simply disable them.
+ */
+public final class GroovyClassValue {
+
+    private GroovyClassValue() {
+    }
+
+    public static void disable() {
+        System.setProperty("groovy.use.classvalue", "false");
```

Comment:
Do you have any references that point to this not being the default value?

Reply:
Yes, you can see that in the source code of Groovy https://github.com/apache/groovy/blob/GROOVY_4_0_X/src/main/java/org/codehaus/groovy/reflection/GroovyClassValueFactory.java#L32

Reply:
Weird how the value of `true` is supposed to prevent a metaspace leak while according to this PR it's the opposite

Reply:
Let's then add some documentation on why this works as I am absolutely sure no one will remember in the future :)

Reply:
I explained everything at the Class definition level, please let me know if it is not enough.

Reply:
Which links are you referring to?

Reply:
Groovy JIRA

Reply:
Should be good now